evaluation.py

```diff
@@ -1,20 +1,38 @@
+import os
+import re
 import cv2
 import matplotlib.pyplot as plt
 import numpy as np
 from PIL import Image
 
 from constants import *
-from HeatMap import HeatMap  # https://github.com/LinShanify/HeatMap
+import HeatMap  # https://github.com/LinShanify/HeatMap
+import util
+import hourglass
+
 
 
 class Evaluation():
 
-    def __init__(self, model_json, weights, df, num_hg_blocks, batch_size=1):
-        self.model_json = model_json              # json of model to be evaluated
-        self.weights = weights          # weights of model to be evaluated
-        self.df = df                    # df of the the annotations we want
-        self.num_hg_blocks = num_hg_blocks
-        self.batch_size = batch_size
+    def __init__(self, model_sub_dir, epoch, model_base_dir=DEFAULT_MODEL_BASE_DIR, output_base_dir=DEFAULT_OUTPUT_BASE_DIR): 
+        # automatically retrieve json and weights
+        self.model_sub_dir=model_sub_dir
+        self.epoch=epoch
+        match = re.match(r'(.*)(_resume_.*$)', model_sub_dir)
+        if match:
+            self.output_sub_dir = os.path.join(output_base_dir, match.group(1), str(self.epoch))
+        else:
+            self.output_sub_dir = os.path.join(output_base_dir, self.model_sub_dir, str(self.epoch))
+        if not os.path.exists(self.output_sub_dir):
+            os.makedirs(self.output_sub_dir)
+
+        self.model_json, self.weights = util.find_resume_json_weights_str(model_base_dir, model_sub_dir, epoch)
```

Comment:
if mine goes in first, you'll need to make this change:
```suggestion
        self.model_json, self.weights, _ = util.find_resume_json_weights_str(model_base_dir, model_sub_dir, epoch)
```